woocommerce/woocommerce-android
# [Tooling] Use a trusted CI agent for pipelines that require push access

### Description
This PR uses the trusted agent helper script `use-bot-for-git` to use the GitHub bot for performing git push operations on CI.

### Testing Steps

We have to time the replacing of the existing deploy key to a read-only one to be able to test this PR, as it would break any ongoing release steps.
I'll add the tests to this PR before the Play Store Submission of version `19.4` (Jul 12th), then cherry-pick the changes to the `release/19.4` branch.

This PR will need anyway to be tested during the next release cycle of the project, when the affected pipelines should still be able to finish successfully.

### Follow-up
Once this PR is merged and making to a code freeze, we will need to revoke and replace in our setup the current deploy key with the new read-only one.
